File: Java/프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.Stack;

class Solution {
    boolean solution(String s) {
        boolean answer = true;
        Stack<Character> stack = new Stack<>();
        char[] arr = s.toCharArray();
        for(char comp : arr){
            if(comp == '('){
                stack.push(comp);
            }else if(!stack.empty() && comp == ')'){
                stack.pop();
            }else{
                stack.push(comp);
            }
        }
        if(!stack.empty()){
            answer = false;
        }

        // [실행] 버튼을 누르면 출력 값을 볼 수 있습니다.
        System.out.println("Hello Java");

        return answer;
    }
}

File: Python3/프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []
    answer = True
    for p in s:
        if p == '(':
            stack.append(p)
        elif p == ')' and len(stack) != 0:
            stack.pop()
        else:
            stack.append(p)
    if len(stack) != 0:
        answer = False
    print(stack)
    
    
    # [실행] 버튼을 누르면 출력 값을 볼 수 있습니다.
    # print('Hello Python')

    return answer